components/gitpod-protocol/src/protocol.ts

```diff
@@ -128,6 +162,7 @@ export type IDESettings = {
     // DEPRECATED: Same with useDesktopIde.
     defaultDesktopIde?: string;
     useLatestVersion?: boolean;
+    skipIdeSelection?: boolean;
```

Comment:
I think we should show `code` as the default value on both the `Preference` and `Dashboard` pages, because without actually making any selections, workspaces will start with code. (code as the default option without modifying the user setting in DB)

In this case, when we click `skip`, it is equivalent to the user choosing code by default. So there is no need to add `skipIdeSelection`.

If we want to know how many people chose `skip`, we should add a `track event`

Reply:
We'll be able to know how many users skipped also with the new event `ide_configuration_skipped`.

I am not strongly against saving `defaultIde` in the DB the information, but at the moment I don't see the benefit of doing it? 

If the `defaultIde` is missing, we would default to `code`, as you said. With the benefit that in the DB we can clearly know who has consciously chose `code`.

Perhaps, could you try to elaborate more on why you think we should save this info in the DB? :)

Reply:
> In this case, when we click `skip`, it is equivalent to the user choosing code by default. So there is no need to add `skipIdeSelection`.

Skip means users like our default setting. So we store `code` as default

If not, will user see this prompt again if he refreshes page, 
=> if so, it's not a good experience.
=> if not, what should we start as IDE when user choose to skip and start a workspace?

IMO, `skip` means skip to use default, and don't alert this again.

Reply:
I am assuming that if a user is happy with VS Code, they would click on it :) 

In my mind, "skip" means "I don't want to take a decision now" rather than "I want VS Code"

Reply:
I don't think this is resolved(still thought `skipIdeSelection` is not necessary). unresolved now

Reply:
@mustard-mh I resolved it because there is not going to be A `Skip` button anymore.